runtime/common/src/paras_registrar.rs

```diff
@@ -271,31 +274,52 @@ pub mod pallet {
 		pub fn swap(origin: OriginFor<T>, id: ParaId, other: ParaId) -> DispatchResult {
 			Self::ensure_root_para_or_owner(origin, id)?;
 
+			// If `id` and `other` is the same id, we treat this as a "clear" function, and exit
+			// early, since swapping the same id would otherwise be a noop.
+			if id == other {
+				PendingSwap::<T>::remove(id);
+				return Ok(())
+			}
+
+			// Sanity check that `id` is even a para.
+			let id_lifecycle =
+				paras::Pallet::<T>::lifecycle(id).ok_or(Error::<T>::NotRegistered)?;
+
 			if PendingSwap::<T>::get(other) == Some(id) {
-				if let Some(other_lifecycle) = paras::Pallet::<T>::lifecycle(other) {
-					if let Some(id_lifecycle) = paras::Pallet::<T>::lifecycle(id) {
-						// identify which is a parachain and which is a parathread
-						if id_lifecycle.is_parachain() && other_lifecycle.is_parathread() {
-							// We check that both paras are in an appropriate lifecycle for a swap,
-							// so these should never fail.
-							let res1 = runtime_parachains::schedule_parachain_downgrade::<T>(id);
-							debug_assert!(res1.is_ok());
-							let res2 = runtime_parachains::schedule_parathread_upgrade::<T>(other);
-							debug_assert!(res2.is_ok());
-							T::OnSwap::on_swap(id, other);
-						} else if id_lifecycle.is_parathread() && other_lifecycle.is_parachain() {
-							// We check that both paras are in an appropriate lifecycle for a swap,
-							// so these should never fail.
-							let res1 = runtime_parachains::schedule_parachain_downgrade::<T>(other);
-							debug_assert!(res1.is_ok());
-							let res2 = runtime_parachains::schedule_parathread_upgrade::<T>(id);
-							debug_assert!(res2.is_ok());
-							T::OnSwap::on_swap(id, other);
-						}
-
-						PendingSwap::<T>::remove(other);
-					}
+				let other_lifecycle =
+					paras::Pallet::<T>::lifecycle(other).ok_or(Error::<T>::NotRegistered)?;
+				// identify which is a parachain and which is a parathread
+				if id_lifecycle == ParaLifecycle::Parachain &&
+					other_lifecycle == ParaLifecycle::Parathread
+				{
+					// We check that both paras are in an appropriate lifecycle for a swap,
+					// so these should never fail.
```

Comment:
Maybe beyond the scope of this PR, but the code here and the `else if` below could be made less redundant by factoring out into function e.g.

```rust
fn do_thread_and_chain_swap(to_downgrade: ParaId, to_upgrade: ParaId) {
  				let res1 = runtime_parachains::schedule_parachain_downgrade::<T>(to_downgrade);
					debug_assert!(res1.is_ok());
					let res2 = runtime_parachains::schedule_parathread_upgrade::<T>(to_upgrade);
					debug_assert!(res2.is_ok());
					T::OnSwap::on_swap(to_upgrade, to_downgrade);
}